insights/combiners/mounts.py

```diff
@@ -43,11 +43,14 @@ class Mounts(object):
 
     def __init__(self, binmount, procmounts, mountinfo):
         self._mounts = {}
+        all_mount_points = set()
```

Comment:
I'm not sure I understand this change, this is a combiner that's supposed to combine all the mount points discovered in each parser that exists.  This change is no longer combining the mounts, and is instead only setting all_mount_points to which ever parser is found in an order. If you could explain further the reasoning behind this change it might help me understand.

Also you referenced the gpfs filesystem, but that file system doesn't appear to be supported in RHEL. So I'm not sure why it's being brought into core, but if it needs to be and gpfs mounts are having an issue. I would recommend creating a separate combiner/parser specifically for gpfs mounts. So that it doesn't break any rules that use this combiner currently.

Reply:
The gpfs has internal and external mounts. Below kind of internal mounts are just in /proc/mounts and /proc/self/mountinfo.  These internal mounts are not in the output of command mount.  We do not need to analyze the internal mounts. So the code changes is to use the output of mount only when it is available.  We do not support gpfs, but the gpfs internal mounts impact the combiner for the original code(Many internal mount points will be unioned)


apps_fs /data/ire/apps gpfs rw,seclabel,noatime 0 0
apps_fs /apps/ampl gpfs rw,seclabel,noatime 0 0
apps_fs /apps/anaconda gpfs rw,seclabel,noatime 0 0
apps_fs /apps/Gauss14 gpfs rw,seclabel,noatime 0 0
apps_fs /apps/gauss16 gpfs rw,seclabel,noatime 0 0
apps_fs /apps/GeoDa gpfs rw,seclabel,noatime 0 0
apps_fs /apps/ibm gpfs rw,seclabel,noatime 0 0
apps_fs /apps/intel gpfs rw,seclabel,noatime 0 0
apps_fs /apps/iveware gpfs rw,seclabel,noatime 0 0
apps_fs /apps/matlab gpfs rw,seclabel,noatime 0 0
apps_fs /apps/R gpfs rw,seclabel,noatime 0 0
apps_fs /apps/SAS gpfs rw,seclabel,noatime 0 0
apps_fs /apps/scripts gpfs rw,seclabel,noatime 0 0
apps_fs /apps/shared gpfs rw,seclabel,noatime 0 0
apps_fs /apps/stata16 gpfs rw,seclabel,noatime 0 0

Reply:
One example here, One system with gpfs service running on RHEL linux, there are more than 40000+ items in /proc/mounts, this leads to this combiner costs 25mins to  parse the content.